x-pack/plugins/security_solution/public/common/components/query_bar/index.tsx

```diff
@@ -102,7 +106,26 @@ export const QueryBar = memo<QueryBarComponentProps>(
       [filterManager]
     );
 
-    const indexPatterns = useMemo(() => [indexPattern], [indexPattern]);
+    const dataView = useMemo(() => {
+      if (Object.hasOwn(indexPattern, 'getName')) {
```

Comment:
if `indexPattern` has `getName`, we can be sure it's the instance of `DataView`? If it so, can we use type guard here, instead of returning casted 

```ts
        return [indexPattern] as DataView[];

```